be/src/vec/functions/array/function_array_element.h

```diff
@@ -81,6 +94,79 @@ class FunctionArrayElement : public IFunction {
     }
 
 private:
+    //=========================== map element===========================//
+    ColumnPtr _get_mapped_idx(const ColumnArray& key_column,
+                           const ColumnWithTypeAndName& argument) {
+        if (key_column.get_data().is_column_string()) {
+            return _mapped_key_string(key_column, argument);
```

Comment:
key compare can be unified for all type using column.compare_at

Reply:
I make a _mapped_key function